File: python_pset6/mario.py
# import a library for getting interger
from cs50 import get_int

# prompt user for height
while True:
    h = get_int("Height: ")
    # if height is less than 1 or higher than 8, prompt user for height again
    if h >= 1 and h <= 8:
        break

# define an interger to print one more # each line each time
j = 1
j <= h

# define an interger to print one less space each line each time
k = h - 1
k >= 0

# define the line loop for printing
for i in range(h):
    # print space without new line then print #
    print(" " * k, end="")
    k -= 1
    # print # one more each line
    print("#" * j)
    j += 1

File: Mario_Cash_pset1/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height_int(string prompt);

int main(void)
{
    // Prompt user for height
    int i = get_height_int("Height: ");
    
    for (int j = 1; j <= i; j++)
    {
        // Print left-align space
        for (int k = 1; k <= i - j; k++)
        {
            printf(" ");
        }
        // Print right-align hashes
        for (int k = 1; k <= j; k++)
        {
            printf("#");
        }
        printf("\n");
    }
}

// If height is less than 1 or higher than 8 or not a interger, then prompt height again
int get_height_int(string prompt)
{
    int n;
    do
    {
        n = get_int("%s", prompt);
    }
    while (n < 1 || n > 8);
    return n;
}